backend/drm/drm.c

```diff
@@ -997,6 +1041,17 @@ static void realloc_crtcs(struct wlr_drm_backend *drm, bool *changed_outputs) {
 
 		wlr_output_damage_whole(&conn->output);
 	}
+
+	free(possible_crtc);
+	free(matched);
+	free(crtc);
+	free(crtc_res);
+	free(connectors);
+
+	return;
+allocate_fail:
```

Comment:
Are you suggesting something like:

```c
error_crtc:
	wlr_log(WLR_ERROR, "Failed to allocate memory");
error_connectors:
	free(crtc);
	wlr_log(WLR_ERROR, "Failed to allocate memory");
error_possible_crtc:
	free(connectors);
	free(crtc);
	wlr_log(WLR_ERROR, "Failed to allocate memory");
error_crtc_res:
	free(possible_crtc);
	free(connectors);
	free(crtc);
	wlr_log(WLR_ERROR, "Failed to allocate memory");
error_matched:
	free(crtc_res);
	free(possible_crtc);
	free(connectors);
	free(crtc);
	wlr_log(WLR_ERROR, "Failed to allocate memory");
```

and just in this particular function or any that I've changed?

Reply:
You need more return statements, but yes, something like that. Some of these can be combined into one label, too.

Reply:
Something among the lines of

```c
error_crtc:
	free(crtc);
error_connectors:
	free(connectors);
error_crtc_res:
	free(crtc_res);
error_matched:
	free(matched);
	wlr_log(WLR_ERROR, "Failed to allocate memory");
```

Reply:
Ah! I misunderstood how these work. I didn't think that it would fall through to the next statement even if a `return` was missing. Alright, I will get this done.